metals/src/main/scala/scala/meta/internal/metals/ScalaVersions.scala

```diff
@@ -134,12 +134,16 @@ object ScalaVersions {
       .flatMap { m =>
         val hasUnderscorePrefix = Option(m.group(1)).isDefined
         val major = m.group(2)
-        val minor = m.group(3)
-        val ending = Option(m.group(4)).map(s => s"$s").getOrElse("")
-        val version = s"$major.$minor$ending"
-        if (isSupportedScalaBinaryVersion(version))
-          Some(version -> hasUnderscorePrefix)
-        else None
+        val minor = Option(m.group(3)).fold("")(identity)
+        val ending = Option(m.group(4)).fold("")(identity)
+        val version = s"$major$minor$ending"
+
+        val asBinary = version match {
+          case _ if isSupportedScalaBinaryVersion(version) => Some(version)
+          case "3" => Some("3.0")
```

Comment:
Binary should actually be `3`, where is `3.0` coming from?

Reply:
From buildInfo that uses this function:  https://github.com/scalameta/metals/blob/1dd0b0d9bbc8c0caddd7e156091fb7bc76fcc677/build.sbt#L215-L220

I'm not sure if it should be just "3". Will it be "3" for scala 3.1.0?

Reply:
> Will it be "3" for scala 3.1.0?

@dos65 yes, that's the binary compatibility promise for Scala 3.

Reply:
I love how everyone including me is confused by that :sweat_smile:

Reply:
@gabro thx! updated

Reply:
my greatest OSS contribution recently is to confirm that, yes, it will be `_3` for all 3.x.y 😆